### DISCUSSION TITLE
Confirmed working on...

### DISCUSSION THREAD

**Discussion by 007revad** [007revad/Synology_enable_eunit#24] [General]
Confirmed working on RS1221+ with DX517

**Comment by 007revad** [comment]
Confirmed working on DS1821+ with DX213
Confirmed working on DS720+ with DX213

**Comment by LMQTLV** [comment]
Hello,
I confirme that it works perfectly on an DS720+ with a DX213.
Thank you very much for this script :)
I just hope it holds up to future updates...

**Comment by kunn79** [comment]
Although it's not mentioned, but it seems a DX513 will work fine on a DS718+ or a DS918+ with the script? 
Dear @007revad do you know if this is true?
Thanks!

**Reply by 007revad** [reply]
Unfortunately the script only works on Synology NAS models that use a device tree blob (/etc/model.dtb). Other models like the DS718+ would need a modified Linux kernel.

**Reply by kunn79** [reply]
Aw ... i just bought a used DX513 (USD $200) and it's on the way. I have a DS718+ and a DS918+. Thought it will work on both of them since they are very close - core hardware-wise. 
I was thinking to use the 513 + 718 combo at the office and move the 918 back home for more HDD space...

**Reply by 007revad** [reply]
The DX513 may just work when you plug it in, with a drive in it.

I just searched in DSM 7.2.1 for the DS718 for files containing "dx513" and then "dx517". The same 5 files contain both dx513 and dx517.
```
diskaction.xml
dx513_v7.db
libhwcontrol.so.1
scemd.xml
updater
```
None of the kernel modules contained "dx513" or "dx517" so maybe support is built into DSM.

**EDIT** People here https://community.synology.com/enu/forum/1/post/128556 say the DX513 works with a DS718+ even though Synology Support said it wouldn't.

**EDIT 2** Let me know if it does work, and which DSM version you are using.

**Reply by kunn79** [reply]
Hi Dave, 

Thanks for your quick response and the references!

Good news #1, the used dx513 arrived and looks intact;
Good news #2, my ds718 recogonize it, and the hdds which I just placed in it,  straight away.

Bad news, no matter which bays I inserted the hdds in, it failed either on the creating pool or volume process. 

Errors are mostly like:
"System failed to create [Storage Pool 2](Device Type is [Basic]) with drive [DX513-1: 1]."
or very occationally, DSM stumbled to create a new storage pool, it came to another errorr "System failed to create [Volume 2] on [Storage Pool 2]." 

I tried different HDDs and a SATA SSD, neither worked... restarted both devices a few times, still no luck

The seller told me he used a ds918 with this dx513 and it worked all good. He took videos on the DSM pages before shipped out. 

His specs: ds918 @ DSM 7.2.1 + dx513 with 5 bays fully loaded
Mine: ds718 @ DSM 7.2.1 +  dx513 with 2 bays loaded (I dont have that many spare drives in hand)

 :-( 

![WX20240923-224932@2x](https://github.com/user-attachments/assets/9f40732c-1929-4c21-a8fe-be6f7acdea23)
![WX20240923-224839@2x](https://github.com/user-attachments/assets/12c83d80-7af1-4f64-a6b0-99aecfb83cbb)
![WX20240923-231212@2x](https://github.com/user-attachments/assets/66fc02ec-4b13-4946-be33-e03c19ea119f)

**Comment by 007revad** [comment]
It could be the eSATA cable or the power supply. Try the DX513 with just 1 drive.

If those 3TB drives have been used in a computer you may need to delete all existing partitions that are on them. I usually connect the drive to a PC then open Disk Management and the delete the partitions.

Also see if the DX513 works with a drive that has a storage pool already created: 
1. Shut down the NAS.
2. Unplug the DX513.
3. Remove both 4TB drives from the NAS.
4. Insert one 3TB drive into the NAS.
5. Boot the NAS.
6. Find the NAS with finds.synonology.com or Synology Assistant.
7. Do the initial DSM setup and create a storage pool and volume.
8. Shut down the NAS.
9. Connect the DX513.
10. Move the 3TB drive from the NAS to the DX513.
11. Insert both 4TB drives back into the NAS.
12. Boot the NAS.
13. Check is DSM mounted the 3TB in the DX513.

**Reply by kunn79** [reply]
Thanks man for the warm and detailed help!

Result first: bad news - still failed. 

It started with something very promissing: Because a sufficient number of drives have been detected, you can now click ... > Online Assemble to assemble this storage pool while the system remains powered on.

Then I chose to Assemble them, and 5 munites later...

Similar bad thing happened. I'm about to move these two drives and the DX513 to the office and see if it works fine with the ds918+. 

![QQ20240925-085823](https://github.com/user-attachments/assets/ef126c3b-b51c-4cfb-bab6-b504523e548f)

![QQ20240925-085846](https://github.com/user-attachments/assets/9ec6074b-ab4d-434c-9f92-2d26a0844005)

![QQ20240925-090715](https://github.com/user-attachments/assets/85f53c19-9d15-4bea-ac13-9e4c6684fb01)

![QQ20240925-090820](https://github.com/user-attachments/assets/647462eb-f4b1-453d-bc8b-937e2f3ffbe9)

**Reply by kunn79** [reply]
Hi Dave, 

Finally, the dx513 works on the DS918+ straight away, except that the write speed went down to 55m from 110m after a few seconds both copying from Window or internally from the NAS to dx513 with the webgui, if the Raid was set as 1 or SHR. Meanwhile, the read speed was full at 110. 

If I changed to Basic or Raid 0, it worked perfectly at 110m. I don't have enough drives in hand so I cannot test Raid5. 

The DSM is v7.1.1 on the ds918+.

Do you think it's worth to try 7.1.1 on my 718+ (now it's 7.2.1) with the dx513?
I will try your script on 718+ afterwards, too.

**Reply by 007revad** [reply]
First I would try the dx513 on the DS718+ again, now that the storage pool and volume has already been created. 
1. Shut down the DS918+.
2. Unplug the DX513 and connect it to the DS718+.
3. See if DS718+ can now do an assemble without errors.

If you are capable of downgrading the DS718+ to DSM 7.1.1 then I'd be curious to know the DX513 then works.

I'll compare the DS918+'s DSM 7.1.1 to 7.2.1 to see if I can find any extra references to DX513 in 7.1.1

**Reply by 007revad** [reply]
There is no difference between the DS718+'s DSM 7.1.1 and 7.2.1 relating to dx513. So I don't DSM 7.1.1 is going to help.

Maybe there's dust in the 718's eSATA port?

**Reply by kunn79** [reply]
(my reply is gone? let me do it again)

> Maybe there's dust in the 718's eSATA port?

That I don't think so - I have a one-bay eSATA external drive running on the DS718+ for like 2 years with no issues.

> I'll compare the DS918+'s DSM 7.1.1 to 7.2.1

I can't do it for security reason - the DS918+ has GitLab, SVN, docker, and a few more serivices running for our office environment. It's better not to upgrade the DSM to 7.2.1 since Docker will be upgraded to Container. 

My original plan was to migrate everything from the DS918+ to the DS718+/DX513 combo and bring it to the office, so that I will have higher drive capacity on both sites.

I will try to do these then: 
1. Try an assemble on the DS718+ as you suggested;
2. Try with another eSATA cable (non-Synology);
3. Downgrade DS718+ to 7.1.1 and see;
4. Try your script.

One side-question though - is it common for a DX device to have this issue:?
> write speed went down to 55m from 110m after a few seconds both copying from Window or internally from the NAS to dx513 with the webgui, if the Raid was set as 1 or SHR. Meanwhile, the read speed was full at 110.

Many thanks!

**Reply by kunn79** [reply]
I just paypaled you with 2 coffee for all the warm-hearted helps!

**Reply by 007revad** [reply]
> write speed went down to 55m from 110m after a few seconds both copying from Window or internally from the NAS to dx513 with the webgui, if the Raid was set as 1 or SHR. Meanwhile, the read speed was full at 110.

That is fairly normal for RAID 1 or SHR with 2 drives, where write speeds are half the read speed. The write speed starts off fast for a few seconds until the NAS' memory fills up or the drives' write caches fill up then it slows down. With 3 drives in RAID 5 or SHR you'd get the 100MB/s writes, the same as RAID 0.

> I just paypaled you with 2 coffee for all the warm-hearted helps!

Thank you 😃

**Reply by kunn79** [reply]
Just out of curiosity, with all the "newer" DS models, why 718+ isn't supported by the enable eunit hi-tech?

**Reply by kunn79** [reply]
> Unfortunately the script only works on Synology NAS models that use a device tree blob (/etc/model.dtb). Other models like the DS718+ would need a modified Linux kernel.

Sorry you explained the reason, on day 1. 
: p

**Reply by kunn79** [reply]
After a few more tests, I can confirm:
1) It works with a 3rd party cable, on the DS718+. 
2) i can't downgrade the DSM from 7.2 to 7.1. 

How weird it is...

**Reply by 007revad** [reply]
Officially you can't downgrade DSM. Unofficially you can, but it's not easy and can kill the NAS.

As long as the DS718+ works with the 3rd party eSATA cable I would not downgrade DSM.

**Comment by 0nl0gisch** [comment]
After some struggling i can confirm that a DX213 is working with a DS723+

![image](https://github.com/user-attachments/assets/cc3e7945-6d30-4394-af24-58387c2832c5)

![image](https://github.com/user-attachments/assets/b425feb5-5d58-40c2-81b8-8151abb943a5)

**Comment by Crashoveride9579** [comment]
Works on 923+ with DX513
![image](https://github.com/user-attachments/assets/99b4c97b-c289-4528-9e1a-336322ae8ab9)

**Comment by mhenigma** [comment]
Confirmed working with RS1221RP+ and DX517

I cannot thank you enough because I was honestly considering purchasing the RX418 and it pained me when I already have two DX517s. I was going to pay more money to hold 1 less drive, what was I honestly thinking.

**Seeing this modification sealed the deal on upgrading to the RS after 8 great years with my DS1517+ and 7 years with my DS1618+. (You hear that Synology? I bought another $1500+ piece of equipment from you because I knew I would be able to use my existing hardware)** Both still work great and I still use the DS1618+ with a DX517 full of 8TB drives for my cold storage.

Really appreciate your work on this and fingers crossed the Synology doesn't go all Synology and break this. We are still purchasing their products and aren't going to them for support when we go outside normal use, but to limit what compatible expansion bays could be use with what DSs was quite a bit too strong in my opinion.

Ohh, and where is your donation link? I have to send you at least some $$ for the money saved on not having to use the RX418.
